spec/facter/facts/linux/hypervisors/kvm_spec.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+describe Facts::Linux::Hypervisors::Kvm do
+  subject(:fact) { Facts::Linux::Hypervisors::Kvm.new }
+
+  describe '#call_the_resolver' do
+    context 'when hypervisor is virtualbox' do
+      before do
+        allow(Facter::Resolvers::Linux::DmiBios).to receive(:resolve).with(:product_name).and_return('VirtualBox')
+      end
+
+      it 'calls  Facter::Resolvers::Linux::DmiBios' do
+        fact.call_the_resolver
+
+        expect(Facter::Resolvers::Linux::DmiBios).to have_received(:resolve).with(:product_name)
+      end
+
+      it 'have nil value' do
+        expect(fact.call_the_resolver).to be_an_instance_of(Facter::ResolvedFact)
+          .and have_attributes(name: 'hypervisors.kvm', value: nil)
+      end
+    end
+
+    context 'when hypervisor is parallels' do
+      before do
+        allow(Facter::Resolvers::Linux::DmiBios).to receive(:resolve).with(:product_name).and_return('Parallels')
+      end
+
+      it 'calls  Facter::Resolvers::Linux::DmiBios' do
+        fact.call_the_resolver
+
+        expect(Facter::Resolvers::Linux::DmiBios).to have_received(:resolve).with(:product_name)
+      end
+
+      it 'have nil value' do
```

Comment:
has a nil value, this applies also to the other similar tests.

Reply:
changed